damnit/ctxsupport/ctxrunner.py

```diff
@@ -45,6 +47,55 @@ class DataType(Enum):
     Timestamp = "timestamp"
 
 
+class MyMetadataClient:
+    def __init__(self, proposal, timeout=10):
+        self.proposal = proposal
+        self.timeout = timeout
+        self._cache = {}
+
+        proposal_path = extra_data.read_machinery.find_proposal(f"p{proposal:06d}")
+        with open(proposal_path / "usr/mymdc-credentials.yml") as f:
+            document = yaml.safe_load(f)
+            self.token = document["token"]
+            self.server = document["server"]
+
+        self._headers = { "X-API-key": self.token }
+
+    def _run_info(self, run):
```

Comment:
I though this one would be the most important to cache since it's called twice for each run if you want both of sample name and run type (most likely the case).

Reply:
Added in 1d4a877.